cloudinary.go

```diff
@@ -278,24 +286,32 @@ func (r *Cloudinary) Put(file, content string) error {
 		return err
 	}
 	_, err = r.instance.Upload.Upload(r.ctx, tempFile.Name(), uploader.UploadParams{
-		PublicID:       r.getPublicId(file),
+		PublicID:       file,
 		UseFilename:    api.Bool(true),
 		UniqueFilename: api.Bool(false),
-		ResourceType:   r.getResourceType(file),
+		ResourceType:   "auto",
 	})
 	return err
 }
 
 // PutFile stores a new file on the disk.
 func (r *Cloudinary) PutFile(path string, source filesystem.File) (string, error) {
-	return r.PutFileAs(path, source, source.File())
+	uploadResult, err := r.instance.Upload.Upload(r.ctx, source.File(), uploader.UploadParams{
+		Folder:         validPath(path),
+		UseFilename:    api.Bool(true),
+		UniqueFilename: api.Bool(false),
+	})
+	if err != nil {
+		return "", err
+	}
+	return uploadResult.PublicID, nil
 }
 
 // PutFileAs stores a new file on the disk.
 func (r *Cloudinary) PutFileAs(path string, source filesystem.File, name string) (string, error) {
 	uploadResult, err := r.instance.Upload.Upload(r.ctx, source.File(), uploader.UploadParams{
 		Folder:         validPath(path),
-		PublicID:       r.getPublicId(name),
+		PublicID:       name,
```

Comment:
Q: Can the name contain extension?

Reply:
I think we'd better follow the cloudinary suggestion.

Reply:
> I think we'd better follow the cloudinary suggestion.

Yeah, we are following cloudinay.